com.unity.cinemachine/Editor/Upgrader/UpgradeObjectToCm3.cs

```diff
@@ -86,8 +87,17 @@ public GameObject UpgradeComponents(GameObject go)
                      orbital.UpgradeToCm3(go.GetComponent<CinemachineOrbitalFollow>());
                      ConvertInputAxis(go, "Look Orbit X", ref orbital.m_XAxis, ref orbital.m_RecenterToTargetHeading);
                 }
+
                 if (ReplaceComponent<CinemachineTrackedDolly, CinemachineSplineDolly>(go))
-                    go.GetComponent<CinemachineTrackedDolly>().UpgradeToCm3(go.GetComponent<CinemachineSplineDolly>());
+                {
+                    var obsoleteDolly = go.GetComponent<CinemachineTrackedDolly>();
+                    var splineDolly = go.GetComponent<CinemachineSplineDolly>();
+                    obsoleteDolly.UpgradeToCm3(splineDolly);
+                    
+                    // set splineDolly spline reference
+                    if (obsoleteDolly.m_Path != null) 
+                        obsoleteDolly.m_Path.TryGetComponent(out splineDolly.Spline);
```

Comment:
Should do it in obsoleteDolly.UpgradeToCm3(), same way that dollycart does it, above

Reply:
Yes